src/persistence/adapters/mongoose-adapter.ts

```diff
@@ -0,0 +1,57 @@
+import { PersistenceAdapter } from '../persistence-adapter';
```

Comment:
remove persistence/adapters from src. lets create mongoose, prisma, typeorm adapters under examples - and show them as examples, but we do not want them inside the library since they are out of scope. 

Keep just the persistance interface in the library and the implementation under index.ts

Also, the persistence adapter interface should move to index and b e exported from it.